app/models/inventory.rb

```diff
@@ -91,8 +91,8 @@ def participant_count
     participants.count
   end
 
-  def analysis_count
-    Analysis.where(inventory_id: id).count
+  def current_analysis
+    analyses.last
```

Comment:
Is there a guarantee of order on how these come in?  To my mind I believe these are indeterminate so this may not always pull in the last element.